### PR TITLE
feat: basket window integration

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

+ backporting `basket.date_criteria` from `master` (#744)
+ updating validation rules.
+ added tests

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by aaronc — APPROVED** [review]

File: x/ecocredit/basket/msg_create.go [hunk 1]
```diff
@@ -78,3 +78,21 @@ func (m MsgCreate) Route() string { return sdk.MsgTypeURL(&m) }
 
 // Type Implements LegacyMsg.
 func (m MsgCreate) Type() string { return sdk.MsgTypeURL(&m) }
+
+func validateDateCriteria(d *DateCriteria) error {
+	if d == nil {
+		return nil
+	}
+	if x := d.GetMinStartDate(); x != nil {
+		if x.Seconds < -2208992400 { // batch older than 1900-01-01 is an obvous error
```
Comment (ryanchristo) [inline x/ecocredit/basket/msg_create.go:87]:
```suggestion
		if x.Seconds < -2208992400 { // batch older than 1900-01-01 is an obvious error
```

File: x/ecocredit/basket/msg_create_test.go [hunk 2]
```diff
@@ -26,45 +26,103 @@ func TestMsgCreateValidateBasic(t *testing.T) {
...
+		{"allowed_classes-2", MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, DateCriteria: start, AllowedClasses: []string{"class1", ""}},
 			"allowed_classes[1] must be defined"},
-		{MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, MinStartDate: start, AllowedClasses: classes, Fee: sdk.Coins{sdk.Coin{Denom: "1a"}}},
+		{"fee-1", MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, DateCriteria: start, AllowedClasses: classes, Fee: sdk.Coins{sdk.Coin{Denom: "1a"}}},
 			"invalid denom"},
-		{MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, MinStartDate: start, AllowedClasses: classes, Fee: sdk.Coins{sdk.Coin{"aa", sdk.NewInt(-1)}}},
+		{"fee-2", MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, DateCriteria: start, AllowedClasses: classes, Fee: sdk.Coins{sdk.Coin{"aa", sdk.NewInt(-1)}}},
```
Comment (ryanchristo) [inline x/ecocredit/basket/msg_create_test.go:71]: It would be nice to use consistent formatting and to make these cases more readable.

**Review by ryanchristo — APPROVED** [review]
Looks good to me.

File: x/ecocredit/basket/msg_create_test.go (hunk 2 above)
Comment (robert-zaremba) [reply x/ecocredit/basket/msg_create_test.go:71]: yes, will move them to a new line

**Pull request merged**